chain/jsonrpc-primitives/src/errors.rs

```diff
@@ -59,18 +76,67 @@ impl RpcError {
     }
     /// Create an invalid request error.
     pub fn invalid_request() -> Self {
-        RpcError::new(-32_600, "Invalid request".to_owned(), None)
+        RpcError {
+            code: -32_600,
+            message: "Invalid request".to_owned(),
+            data: None,
+            error_struct: Some(RpcErrorKind::ValidationError(ValidationErrorKind::InvalidRequest)),
+        }
     }
     /// Create a parse error.
     pub fn parse_error(e: String) -> Self {
-        RpcError::new(-32_700, "Parse error".to_owned(), Some(Value::String(e)))
+        RpcError {
+            code: -32_700,
+            message: "Parse error".to_owned(),
+            data: Some(Value::String(e.clone())),
+            error_struct: Some(RpcErrorKind::ValidationError(ValidationErrorKind::ParseError {
+                error_message: e,
+            })),
+        }
     }
     pub fn serialization_error(e: String) -> Self {
-        RpcError::new(-32_000, "Server error".to_owned(), Some(Value::String(e)))
+        // handler error
+        RpcError {
+            code: -32_000,
+            message: "Server error".to_owned(),
+            data: Some(Value::String(e.clone())),
+            error_struct: Some(RpcErrorKind::HandlerError(serde_json::json!({
+                "name": "SERIALIZATION_ERROR",
+                "info": serde_json::json!({
+                    "error_message": e
+                })
+            }))),
```

Comment:
Let's not introduce weird corner cases. Let's define `InternalError` variant next to `ValidationError` and `HandlerError`. We may also consider to map `RpcErrorKind -> HandlerError -> * -> InternalError` (just a pseudocode) into `RpcErrorKind -> InternalError`.

I also want to bring another idea here: consider also merging `Rpc*::Unreachable` into `Rpc*::InternalError` (on the RPC error types, not the view-client error types), so end-users don't need to handle two similar types and rather handle Internal Errors as something being wrong on the server side and the only thing client can do is to retry.